sqlglot/dialects/dialect.py

```diff
@@ -469,6 +469,9 @@ class Dialect(metaclass=_Dialect):
     NUMBERS_CAN_BE_UNDERSCORE_SEPARATED = False
     """Whether number literals can include underscores for better readability"""
 
+    HEX_STRING_IS_INTEGER_TYPE: t.Optional[bool] = None
+    """Whether hex strings such as x'CC' evaluate to integer or binary/blob type"""
```

Comment:
Let's make this a simple `bool`, default to `False` and in the parser you can do `<flag> or None` if you want to avoid setting the arg.